accepted/2021/default_string_globalization.md

```diff
@@ -0,0 +1,390 @@
+
+## Characteristics of an ideal solution
+
+An ideal solution to this problem should have two qualities.
+
+1. `string` and related APIs should default to ordinal behavior unless the caller makes an explicit request for a different behavior.
+
+2. Developers should not be exposed to any globalization-related concepts unless they explicitly seek globalization support.
+
+The first characteristic is fairly straightforward. APIs like `string.Compare`, `string.Contains`, `string.Equals`, `string.IndexOf`, and others should all behave in an ordinal fashion. Developers shouldn't have to rely on Roslyn analyzers or MSDN to know what behavior a given API has.
+
+The second characteristic requires a bit more explanation. Popular programming languages like Java, JavaScript, Go, Python, C/C++, and others treat string APIs as ordinal. .NET is unique in that we build globalization concepts directly into many basic `string` APIs. Unfortunately, evidence shows that .NET's behavior in this regard is largely unexpected and confusing to our developer audience.
+
+[Java 16's `String` class](https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/String.html) (which is the closest to .NET's `string` class) is a good example of this principle. Per their own documentation:
+
+> Unless otherwise noted, methods for comparing Strings do not take locale into account.
+
+And in fact, the _only_ two Java `String` APIs that are culture-aware are case-conversion routines.
+
+```java
+String s = getString();
+
+// parameterless overloads use current locale by default
+String s1 = s.toUpperCase();
+String s2 = s.toLowerCase();
+
+// or you can pass a Locale instance if you don't want to use the current locale
+// Java's Locale.ROOT is roughly equivalent to .NET's CultureInfo.InvariantCulture
+String s3 = s.toUpperCase(Locale.ROOT);
+String s4 = s.toLowerCase(Locale.ROOT);
+```
+
+And even though these are the only two linguistic-by-default APIs on Java's `String` type, problematic usage of them is still so widespread that CERT's coding rules for Java [forbid calling the parameterless overloads](https://wiki.sei.cmu.edu/confluence/display/java/STR02-J.+Specify+an+appropriate+locale+when+comparing+locale-dependent+data). This CERT rule STR02-J is roughly equivalent to .NET's [CA1310 rule](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1310), which flags certain overloads of `string` APIs as problematic.
+
+Put another way, the second characteristic mentioned above might be paraphrased as "developers shouldn't be exposed to globalization-related concerns unless they have imported the _System.Globalization_ namespace into their file."
+
+## Proposed solution
+
+To solve this problem for the .NET ecosystem at large, we propose to take the following actions on `string`.
+
+1. Obsolete (as warning) all linguistic-by-default methods on `string` with a new __SYSLIBxxxx__ identifier.
```

Comment:
I'd really like to see 1. be `Create fixers to migrate call sites` and 2. to be `Obsolete (as warning)`

Having a fixer is critical IMO and the most important part considering how widespread the existing methods are.